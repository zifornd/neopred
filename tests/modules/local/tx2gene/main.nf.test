nextflow_process {

    name "Test Process TX2GENE"
    script "modules/local/tx2gene/main.nf"
    process "TX2GENE"

    test("Should run without failures") {

        when {
            process {
                """
                // define inputs of the process here. Example:

		 input[0] = Channel.fromPath("$projectDir/tests/test_result/tx2gene/quants/SRR8281226")
		 //input[0] = Channel.of([['SRR8281243'], file("$projectDir/tests/test_result/tx2gene/quants/")])
                 input[1] = "salmon"
                 input[2] = file("$projectDir/tests/test_result/tx2gene/gencode.v45.primary_assembly_chr6.annotation.gtf")


                """
            }
        }

        then {
            assert process.success
	    assert process.out.tsv
            assert snapshot(process.out)//.match()
        }

    }

}
