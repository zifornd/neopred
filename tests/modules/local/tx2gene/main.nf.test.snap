{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "tx2gene.tsv:md5,b860b65d41b6a14aa56dbe89454674ff"
                ],
                "1": [
                    "versions.yml:md5,b5ca97b1a7c44bb992aa3d1044c3923f"
                ],
                "tsv": [
                    "tx2gene.tsv:md5,b860b65d41b6a14aa56dbe89454674ff"
                ],
                "versions": [
                    "versions.yml:md5,b5ca97b1a7c44bb992aa3d1044c3923f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-12-23T08:32:08.33553"
    }
}